apps/condo/domains/meter/schema/MeterReportingPeriod.js

```diff
@@ -0,0 +1,169 @@
+                        if (resolvedData.start > 31 || resolvedData.start < 1) {
+                            throw new GQLError(ERRORS.INVALID_START, context)
+                        }
+                    }
+                },
+            },
+        },
+
+        finish: {
+            schemaDoc: 'This is the field for client-side business logic and push notifications',
+            type: Integer,
+            defaultValue: 25,
+            isRequired: true,
+            hooks: {
+                validateInput: async ({ context, operation, resolvedData }) => {
+                    if (operation === 'create' || operation === 'update') {
+                        if (resolvedData.finish > 31 || resolvedData.finish < 1) {
+                            throw new GQLError(ERRORS.INVALID_FINISH, context)
+                        } 
+                    }
+                },
+            },
+        },
+
+        b2cApp: {
+            schemaDoc: 'Ref to the B2CApp which used to replace default integration in resident\'s app',
+            type: Relationship,
+            ref: 'B2CApp',
+            isRequired: false,
+            knexOptions: { isNotNullable: false }, // Relationship only!
+            kmigratorOptions: { null: true, on_delete: 'models.SET_NULL' },
+        },
+    },
+    kmigratorOptions: {
+        constraints: [
+            {
+                type: 'models.UniqueConstraint',
+                fields: ['organization', 'property'],
+                condition: 'Q(deletedAt__isnull=True)',
+                name: 'meterreportingperiod_unique_organization_property',
+            },
+            {
+                type: 'models.UniqueConstraint',
+                fields: ['organization'],
+                condition:
+                    'Q(deletedAt__isnull=True) &' +
+                    'Q(property__isnull=True)',
+                name: 'meterreportingperiod_unique_organization',
+            },
+        ],
+    },
+    hooks: {
+        resolveInput: async ({ context, operation, existingItem, resolvedData }) => {
+            if (operation === 'create') {
+                if (!resolvedData.organization && resolvedData.property) {
+                    const property = await Property.getOne(context, { id: resolvedData.property })
+
+                    resolvedData.organization = property.organization
+                }
+            }
+
+            return resolvedData
+        },
+        validateInput: async ({ operation, resolvedData, context }) => {
+            if (operation === 'create') {
+                const foundDefaultPeriod = await MeterReportingPeriodAPI.getAll(context, {
+                    organization_is_null: true,
+                    deletedAt: null,
+                })
+
+                if (!resolvedData.organization && foundDefaultPeriod.length > 0) {
+                    throw new GQLError(ERRORS.ORGANIZATION_IS_REQUIRED, context)
+                }
+            }
+        },
```

Comment:
Can this be done with a constraint like with property in 115-122 lines?

Reply:
I didn't manage to do it using constraint.
You need a record in the database that will be available to all users and use its settings as default. Therefore, a proposal was made to make a record not tied to an `organization`. To do this, I check that there is no such entry, and if there is, I think that the user wanted to create a period for his `organization` and for some reason it is not specified. If you use `constraint`, then you will not be able to create a record that is common to all. I did not find a way that would allow the `constraint` to fire only if there are no records with an empty `organization` field. It seems to me that there should be some kind of `count`, but I did not find it